src/context.cpp

```diff
@@ -483,6 +484,102 @@ namespace librealsense
         query_option_callback _query_opt_cb;
     };
 
+
+    // Frame data and metadata are sent as two seperate streams.
+    // They are synchronized using frame_id for matching data+metadata sets.
+    template< class T >
+    class frame_metadata_syncer
```

Comment:
Can we put this in a separate header?

Reply:
Yes. In the next PR.
I mentioned this at the description, I want to take all of the DDS classes to different files, but I didn't do it in this PR so it would be easier to compare the changes for metadata support.